src/axom/sidre/spio/IOManager.hpp

```diff
@@ -224,7 +226,11 @@ class IOManager
                                      const std::string& mesh_path);
 
   /*!
-   * \brief read from input files
+   * \brief read from input file
+   *
+   * This is an MPI collective call.  Calling code may also need to add
```

Comment:
@nselliott , @cyrush, could you help me understand this use case a little better. 

- Why would the application need to put a barrier after calling `read()` ?  
- Should we also have a barrier at the end of the `read()` (see my earlier comment)?

Reply:
hmm, I don't think we need one after read

Reply:
The only reason a barrier might be needed after read() would be if the application immediately starts doing I/O on the same files right after it gets out of read().  So one rank could complete read() and then start doing something (probably not axom-based) that messes with the files, affecting another rank that is still in read().  So we could put a barrier at the end of read() to protect against that, or we can trust applications to be smart enough to add their own barriers before starting new I/O.  Putting the barrier in our code would add extra latency to applications that want each rank to finish reading and go straight to computing.

Reply:
Thanks -- I understand both use cases.

So, maybe we can have another method, e.g., `readAll()` that basically calls `read()` and adds the barrier synchronization at the end. 

That way we retain the flexibility when needed and it is more clear in application code what is happening. Thoughts?

Reply:
Good suggestion @gzagaris !